File: app/src/main/java/com/amperas17/mangaedenapp/ui/mangafull/view/ChapterAdapter.java
package com.amperas17.mangaedenapp.ui.mangafull.view;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.amperas17.mangaedenapp.R;
import com.amperas17.mangaedenapp.model.chapter.Chapter;
import com.amperas17.mangaedenapp.utils.DateUtils;

import java.util.ArrayList;
import java.util.Date;


class ChapterAdapter extends RecyclerView.Adapter<ChapterAdapter.ViewHolder> {

    ArrayList<Chapter> chapterList;
    private OnItemClickListener listener;

    ChapterAdapter(OnItemClickListener listener) {
        this.chapterList = new ArrayList<>();
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_chapter, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, final int position) {
        holder.bind(chapterList.get(position), listener);
    }

    @Override
    public int getItemCount() {
        return chapterList.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        private TextView tvMangaTitle;
        private TextView tvChapterTitle;
        private TextView tvDate;

        private ViewHolder(final View itemView) {
            super(itemView);
            tvMangaTitle = itemView.findViewById(R.id.tvMangaTitle);
            tvChapterTitle = itemView.findViewById(R.id.tvChapterTitle);
            tvDate = itemView.findViewById(R.id.tvDate);
        }

        private void bind(final Chapter chapterItem, final OnItemClickListener listener) {
            tvMangaTitle.setText(String.valueOf(chapterItem.getNumber()));
            tvChapterTitle.setText(chapterItem.getTitle());
            tvDate.setText(DateUtils.DATE_FORMAT.format(new Date(DateUtils.MILLIS_IN_SECOND * chapterItem.getDate())));

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    listener.onItemClick(chapterItem);
                }
            });
        }
    }

    interface OnItemClickListener {
        void onItemClick(Chapter chapter);
    }
}

File: app/src/main/java/com/amperas17/mangaedenapp/ui/mangafull/view/ChapterAdapter.kt
package com.amperas17.mangaedenapp.ui.mangafull.view

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.amperas17.mangaedenapp.R
import com.amperas17.mangaedenapp.model.chapter.Chapter
import com.amperas17.mangaedenapp.utils.AdapterItemClickListener
import com.amperas17.mangaedenapp.utils.formatDefaultFromSeconds
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_chapter.*

import java.util.ArrayList


class ChapterAdapter(private val listener: AdapterItemClickListener<Chapter>) : RecyclerView.Adapter<ChapterAdapter.ViewHolder>() {

    var chapterList: ArrayList<Chapter> = ArrayList()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {

        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.item_chapter, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(chapterList[position], listener)
    }

    override fun getItemCount(): Int {
        return chapterList.size
    }

    inner class ViewHolder(override val containerView: View) : RecyclerView.ViewHolder(containerView), LayoutContainer {

        fun bind(chapterItem: Chapter, listener: AdapterItemClickListener<Chapter>) {
            tvMangaTitle.text = chapterItem.number.toString()
            tvChapterTitle.text = chapterItem.title
            tvDate.text = chapterItem.date.formatDefaultFromSeconds()

            itemView.setOnClickListener { listener.onItemClick(chapterItem) }
        }
    }
}
